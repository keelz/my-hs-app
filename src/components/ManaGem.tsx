import * as React from 'react';
import * as classnames from 'classnames';

export interface IManaGemProps {
  className?: string;
  cost: number;
  onClick?: () => void;
}

const composeClassnames = (className?: string) =>
  classnames(
    [
      'btn',
      'btn-primary',
      'Mana-gem',
    ],
    className
  );

const ManaGem: React.SFC<IManaGemProps> = props =>
  <button
    className={composeClassnames(props.className)}
    onClick={props.onClick}>
    { props.cost }
  </button>;

export default ManaGem;
