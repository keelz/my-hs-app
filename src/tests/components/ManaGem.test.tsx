import * as React from 'react';
import * as enzyme from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ManaGem from '../../components/ManaGem';

describe('ManaGem', () => {
  describe('snapshots', () => {
    it('renders without crashing', () => {
      const tree = enzyme.shallow(<ManaGem cost={2} />);
      expect(shallowToJson(tree)).toMatchSnapshot();
    });
  });
});
