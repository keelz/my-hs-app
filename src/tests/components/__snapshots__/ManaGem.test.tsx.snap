// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManaGem snapshots renders without crashing 1`] = `
<div
  className="Mana-gem"
>
  <img
    alt="gem"
    className="Mana-gem-nav-img"
  />
  <div
    className="Mana-gem-nav-text"
  >
    2
  </div>
</div>
`;
